### PR TITLE
feat: add android legacy compatibility check

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [ ] PR title descriptive (can be used in release notes)

## Description
Add Android compatibility check to prompt users to upgrade their underlying google chrome or android webview applications if using outdated versions (pre March 2021)

## Author Notes
We're following Capacitor's advice to target API 23 and higher, but if useful we could manually specify API 22 so that play store will allow installation and our error message will show instead. I assume we probably want to keep as-is, so that the play store simply shows incompatibility message and prevents install (may be better localised)

As mentioned on the call, the messages displayed can only be localised as far as us providing different strings and switching based on user device language which might be a little complicated to manage given the number of languages we work with and fact most users don't update their operating system language (limited options plus technical challenges to do so).

I've included some emojis in the text to try and give some better visual clues of text, but otherwise the next best option would likely be scripts that change the text as a one-time replacement as part of android workflow for deployments. That is beyond the scope of this PR, but could be a follow-up issue.

## Review Notes
See screenshots below of various android emulators for different Android versions.
If testing locally use the `Show obsolete device profiles` checkbox to access devices capable of running Android 5 or 6

If wanting to test app following chrome/webview update, will additionally need to include an image with Google Play Store enabled from the services (may not exist for all android versions), and sign in on device. Will also need to build and run `yarn workflow android` to sync updated android template files

image

## Dev Notes
I had originally planned to include as JS code to be evaluated from `index.html`, however it appears that in some more recent capacitor updates they fail to render any UI if chrome version deemed incompatible. This meant a white screen would show on Android 7, still executing the check but failing to render an alert. As such I've rewritten the check in Java and run alongside the main app startup code (I had tried to block app startup pending compatibility pass however this would throw error as it always expects the main activity create method to be called). I think it makes sense as it's mostly android compatibility we care about (legacy chrome web users likely to be forced to update when using browser)

There might be one edge case if an Android 6 user has manually installed chrome but has never updated webview the app still won't work but update prompt will be for chrome (not webview) - but I'm not sure that's really worth handling separately (logic already getting a bit complex). There's also an edge case in Android 7 where two different webview apps might be available depending on manufacturer - I've done my best to capture but unable to test without different devices (AI suggests that different names of `com.google.android.webview` and `com.android.webview` that both update from same play store URL)

The minimum version of chrome required is somewhat customisable. By default angular will look at the `.browserslistrc` file and extract target chrome version from there. This is currently hardcoded at `chrome 89`, likely due to legacy ionic recommendations. Angular will automatically add polyfills for any apis the app uses that are not available in minimum target version (e.g. Promise.all, most es2020 features etc.). Whilst a lower target means fewer devices will get an upgrade prompt, it does result in larger builds (currently polyfills sit around 150kb) and there may also be some performance implications.

I think given general bundle sizes being quite large this isn't a huge issue, but if we ever want a better-optimised build for web would definitely be worth reviewing. 

## Git Issues

Closes #

## Screenshots/Videos
User experience for devices by android version, assuming user has never updated webview/chrome app (new device)

| API | Android | Emulator Chrome Version | User Experience | Screenshot | 
| ---|----|---|---|---|
|22 | 5.1 | NA | App fails to install | image |
|23 | 6.0 | 44 | Splash screen launches, use sees update message on top of white screen | image |
|25 | 7.1 | 55 (via webview) | Splash screen launches, user sees update message on top of white screen | image |
| 30 | 11.0 | 83 (via webview) | App launches, user sees update message on top of content | image |
 | 31 | 12.0 | 91 | App runs as normal | image |

**Demo - Android 7 message, update and working**

[untitled.webm](https://github.com/user-attachments/assets/07b5a530-a2af-4549-8383-ed332e1f4b9d)

### CODE REVIEW

File: android/app/src/main/java/utils/WebViewChecker.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    private static int getMajorVersion(String versionName) {
+        if (versionName == null) return 0;
+        try {
+            String[] parts = versionName.split("\\.");
+            return Integer.parseInt(parts[0]);
+        } catch (Exception e) {
+            return 0;
+        }
```
[the remark this answers is not in the document]
Comment (chrismclarke) [reply android/app/src/main/java/utils/WebViewChecker.java:97]: Not important as only used to extract version number

File: android/app/src/main/java/utils/WebViewChecker.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    private static void showUnsupportedDialog(Activity activity, String message) {
+        new AlertDialog.Builder(activity)
+            .setTitle("Unsupported Device")
+            .setMessage(message)
+            .setCancelable(false)
+            .setPositiveButton("Exit", (dialog, which) -> activity.finish())
+            .show();
+    }
```
[the remark this answers is not in the document]
Comment (chrismclarke) [reply android/app/src/main/java/utils/WebViewChecker.java:107]: will likely need to expose to replacement/templating system in the future, so can manage then

**Comment by chrismclarke** [comment]
> I was surprised to see that the itemised list in the update message is right-aligned in your screenshots (but the text above isn't), it would make more sense to me if it were left aligned. But if not straightforward to resolve or if you think this is preferable we can leave as is.

That comes from the Android system level so not something in our control (notification system buttons are mostly right-aligned, although might vary a bit by android version). So yeah I think this is the best we can currently do
